simln-lib/src/sim_node.rs

```diff
@@ -675,6 +682,12 @@ impl<T: SimNetwork> LightningNode for SimNode<T> {
             },
         };
 
+        entry.insert(InFlightPayment {
+            track_payment_receiver: receiver,
+            path: route.paths[0].clone(), // TODO: how to handle non-MPP support (where would we do
```

Comment:
I think that we can generally tighten up handling of paths while we're here. Let's:
- Check in `send_payment` and `send_to_route` that we only have one path
- Have a defensive check in `dispatch_payment` that we only have one path

Can just be a commit in this PR I think!